lib/ex_unit/test/ex_unit/callbacks_test.exs

```diff
@@ -23,13 +23,22 @@ defmodule ExUnit.CallbacksTest do
         {:ok, [context: :setup]}
       end
 
+      setup %{foo: "bar"} do
+        # I should not enter here
+        assert 0 > 1
+      end
+
+      setup %{value: value} = context when is_integer(value) do
+        assert value == 1
+        {:ok, context}
+      end
+
       test "callbacks", context do
         assert context[:context] == :setup
       end
     end
 
-    assert capture_io(fn -> ExUnit.run end) =~
-           "1 test, 0 failures"
+    capture_io(fn -> ExUnit.run end) =~ "1 test, 0 failures"
```

Comment:
I forgot to restore the `assert` after deciding not to change anything here, sorry :sweat: